usr/sbin/pkcstok_migrate/pkcstok_migrate.c

```diff
@@ -1678,10 +1717,8 @@ static CK_RV create_NVTOK_DAT_312(const char *data_store, const char *sopin,
     }
 
     /* Create new file NVTOK.DAT_312 */
-    sprintf(fname, "%s/%s", data_store, nvtok);
-    fp = fopen(fname, "w");
+    fp = open_datastore_file(fname, sizeof(fname), data_store, nvtok, "w");
     if (!fp) {
-        TRACE_ERROR("fopen(%s) failed, errno=%s\n", fname, strerror(errno));
         ret = CKR_FUNCTION_FAILED;
         goto done;
     }
```

Comment:
I would suggest to write a function
inline int __attribute__((format(printf,3,4))) ock_snprintf(buf, buflen, fmt, ...)
{
    va_list ap;
    int n;

    va_start(ap, fmt);
    n = vsnprintf(buf, buflen, fmt, ap);
    va_end(ap);

    if (n < 0 || n >= buflen)
        return -1;
  
    return 0;
}

and use this

if (ock_snprintf(buf, sizeof(buf), "%s %s ...", str1, str2) != 0) {
    TRACE_ERROR("...");
    // do something more
}

Reply:
I don't understand this comment.  What does it refer to?  Error tracing is moved into the open_datastore_file function.  Where would I need this new function?

Reply:
It refers to all the code above. You could declare a new function ock_snprintf like above and use it
everywhere in ock where as of now snprintf is used. It delegates the right return code handling to this
function and the caller does not even need to declare the int snres variable but only check if the returncode
is != 0 for failure.
That's a suggestion I would do - you may have another opinion about this.

Reply:
Okay, this would remove the need for the additional local variable that stores the return value.  However, I think this requires a new header that should probably not be in usr/include, right?

Reply:
Done.  Also simplified one more piece of code (at the cost of an additional local variable).